### DISCUSSION TITLE
[meta] REPL mode

### DISCUSSION THREAD

**Discussion by vovcacik** [junegunn/fzf#2636] [Ideas]
- #1412
- #2465
  - #2643
- #2532
- #2540
- #2628
  - the idea of this one is piping whole buffer through command written by user in query input
- #2633
- #2635

todo #395
todo #2643
todo #2531

**Comment by rayiik** [comment]
hey i recall seeing this in a blog just found it again but 
```bash
$ echo '' | fzf --print-query --preview 'echo "a\nb\nc\nd" | awk {q}'
```
[https://github.com/pawelduda/fzf-live-repl](url)
 
Also Regex is possible 

[![asciicast](https://asciinema.org/a/SvbyryLx0MI6Dy4s1sIOxYDDM.svg)](https://asciinema.org/a/SvbyryLx0MI6Dy4s1sIOxYDDM)

i haven't developed it past concept but the plan was to take whatever successful matching regex pattern i made saving it and for a proper filter later in the script. (also this was based off someone elese's work just cant remember where i picked it up)

```bash
declare gvar=''
declare Fvar=''
if [[ -z "$1" ]]; then
  Fvar='.'; else
  Fvar="$1"
fi
if [[ -z "$2" ]]; then
  gvar='.'; else
  gvar="$2"
fi
grep_flag_available="rg '{q}' ./ --color=always --colors=match:fg:12 --no-filename"
fd -uu -t f "$Fvar" | rg "$gvar" --no-filename | fzf --phony \
  --preview="nvimpager -c {} | $grep_flag_available"
unset gvar
unset Fvar
```
